### ISSUE TITLE
thread `main` panics when `build` is called

### ISSUE THREAD

**Issue by SimonWoodburyForget** [richardanaya/js-wasm#2]
System info:
- cargo-webassembly 0.0.10
- Arch Linux x86_64 
- Kernel 5.9.10-arch1-1 

Attempt at using it from a cargo workspace, with `frontend` being a member:

```
~/Projects/project-dir/frontend $ cargo webassembly build
   Compiling webassembly `frontend` package
thread 'main' panicked at 'could not copy: Os { code: 2, kind: NotFound, message: "No such file or directory" }', /home/simon/.cargo/registry/src/github.com-1ecc6299db9ec823/cargo-webassembly-0.0.10/src/main.rs:171:6
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

Attempt at using it like the documentation said it should be used:

```
~/Projects $ cargo webassembly new test-project
~/Projects $ cd test-project
~/Projects/test-project $ cargo webassembly build
   Compiling webassembly `test-project` package
thread 'main' panicked at 'could not copy: Os { code: 2, kind: NotFound, message: "No such file or directory" }', /home/simon/.cargo/registry/src/github.com-1ecc6299db9ec823/cargo-webassembly-0.0.10/src/main.rs:171:6
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

Full backtrace:

```
stack backtrace:
   0:     0x5566736714a0 - std::backtrace_rs::backtrace::libunwind::trace::h72c2fb8038f1bbee
                               at /rustc/7eac88abb2e57e752f3302f02be5f3ce3d7adfb4/library/std/src/../../backtrace/src/backtrace/libunwind.rs:96
   1:     0x5566736714a0 - std::backtrace_rs::backtrace::trace_unsynchronized::h1e3b084883f1e78c
                               at /rustc/7eac88abb2e57e752f3302f02be5f3ce3d7adfb4/library/std/src/../../backtrace/src/backtrace/mod.rs:66
   2:     0x5566736714a0 - std::sys_common::backtrace::_print_fmt::h3bf6a7ebf7f0394a
                               at /rustc/7eac88abb2e57e752f3302f02be5f3ce3d7adfb4/library/std/src/sys_common/backtrace.rs:79
   3:     0x5566736714a0 - <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt::h2e8cb764b7fe02e7
                               at /rustc/7eac88abb2e57e752f3302f02be5f3ce3d7adfb4/library/std/src/sys_common/backtrace.rs:58
   4:     0x5566736933bc - core::fmt::write::h7a1184eaee6a8644
                               at /rustc/7eac88abb2e57e752f3302f02be5f3ce3d7adfb4/library/core/src/fmt/mod.rs:1080
   5:     0x556673669bb2 - std::io::Write::write_fmt::haeeb374d93a67eac
                               at /rustc/7eac88abb2e57e752f3302f02be5f3ce3d7adfb4/library/std/src/io/mod.rs:1516
   6:     0x5566736738ad - std::sys_common::backtrace::_print::h1d14a7f6ad632dc8
                               at /rustc/7eac88abb2e57e752f3302f02be5f3ce3d7adfb4/library/std/src/sys_common/backtrace.rs:61
   7:     0x5566736738ad - std::sys_common::backtrace::print::h301abac8bb2e3e81
                               at /rustc/7eac88abb2e57e752f3302f02be5f3ce3d7adfb4/library/std/src/sys_common/backtrace.rs:48
   8:     0x5566736738ad - std::panicking::default_hook::{{closure}}::hde0cb80358a6920a
                               at /rustc/7eac88abb2e57e752f3302f02be5f3ce3d7adfb4/library/std/src/panicking.rs:208
   9:     0x556673673558 - std::panicking::default_hook::h9b1a691049a0ec8f
                               at /rustc/7eac88abb2e57e752f3302f02be5f3ce3d7adfb4/library/std/src/panicking.rs:227
  10:     0x556673673f91 - std::panicking::rust_panic_with_hook::h2bdec87b60580584
                               at /rustc/7eac88abb2e57e752f3302f02be5f3ce3d7adfb4/library/std/src/panicking.rs:577
  11:     0x556673673b39 - std::panicking::begin_panic_handler::{{closure}}::h101ca09d9df5db47
                               at /rustc/7eac88abb2e57e752f3302f02be5f3ce3d7adfb4/library/std/src/panicking.rs:484
  12:     0x55667367190c - std::sys_common::backtrace::__rust_end_short_backtrace::h3bb85654c20113ca
                               at /rustc/7eac88abb2e57e752f3302f02be5f3ce3d7adfb4/library/std/src/sys_common/backtrace.rs:153
  13:     0x556673673af9 - rust_begin_unwind
                               at /rustc/7eac88abb2e57e752f3302f02be5f3ce3d7adfb4/library/std/src/panicking.rs:483
  14:     0x556673692421 - core::panicking::panic_fmt::h48c31e1e3d550146
                               at /rustc/7eac88abb2e57e752f3302f02be5f3ce3d7adfb4/library/core/src/panicking.rs:85
  15:     0x556673692243 - core::option::expect_none_failed::h6154dc750ae47ade
                               at /rustc/7eac88abb2e57e752f3302f02be5f3ce3d7adfb4/library/core/src/option.rs:1234
  16:     0x556673537f09 - cargo_webassembly::build_project_in_dir::h838dbead406e9216
  17:     0x556673562f52 - <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll::h92ffafc72346c21e
  18:     0x55667356636d - std::thread::local::LocalKey<T>::with::h4e31b0037d0257af
  19:     0x55667355bb74 - <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll::h2921419d8f859af3
  20:     0x55667356ee03 - async_io::driver::block_on::h6d71b6954b3f5c98
  21:     0x556673566759 - std::thread::local::LocalKey<T>::with::hb0bd46fd4478795c
  22:     0x556673565d50 - std::thread::local::LocalKey<T>::with::h1fd816308d93a5b2
  23:     0x556673536cc5 - async_std::task::builder::Builder::blocking::h10ff00bf5b61a5d0
  24:     0x556673537fd8 - cargo_webassembly::main::hccd0d0ce847cbe5f
  25:     0x556673548c13 - std::sys_common::backtrace::__rust_begin_short_backtrace::he3757adc20188f71
  26:     0x556673548c35 - std::rt::lang_start::{{closure}}::hd11f71d1f10ab0e7
  27:     0x5566736743b7 - core::ops::function::impls::<impl core::ops::function::FnOnce<A> for &F>::call_once::he179d32a5d10d957
                               at /rustc/7eac88abb2e57e752f3302f02be5f3ce3d7adfb4/library/core/src/ops/function.rs:259
  28:     0x5566736743b7 - std::panicking::try::do_call::hcb3d5e7be089b2b4
                               at /rustc/7eac88abb2e57e752f3302f02be5f3ce3d7adfb4/library/std/src/panicking.rs:381
  29:     0x5566736743b7 - std::panicking::try::h7ac93b0cd56fb701
                               at /rustc/7eac88abb2e57e752f3302f02be5f3ce3d7adfb4/library/std/src/panicking.rs:345
  30:     0x5566736743b7 - std::panic::catch_unwind::h7b40e396c93a4fcd
                               at /rustc/7eac88abb2e57e752f3302f02be5f3ce3d7adfb4/library/std/src/panic.rs:382
  31:     0x5566736743b7 - std::rt::lang_start_internal::h142b9cc66267fea1
                               at /rustc/7eac88abb2e57e752f3302f02be5f3ce3d7adfb4/library/std/src/rt.rs:51
  32:     0x556673538012 - main
  33:     0x7f175ca58152 - __libc_start_main
  34:     0x55667352a09e - _start
  35:                0x0 - <unknown>
```

**Comment by gatewaynode** [comment]
I get the same error with the run command on ARM/Manjaro:
```
$ cargo webassembly run
   Compiling webassembly some-wasm package
thread 'main' panicked at 'could not copy: Os { code: 2, kind: NotFound, message: "No such file or directory" }', /home/gatewaynode/.cargo/registry/src/github.com-1ecc6299db9ec823/cargo-webassembly-0.0.10/src/main.rs:171:6
```

**Comment by richardanaya** [comment]
hmm, weird, it is not liking that name for some reason! checking it out

**Comment by richardanaya** [comment]
apparently rust autlo converts to underscore on build

**Comment by richardanaya** [comment]
pushed fix!

**Comment by gatewaynode** [comment]
Updated and tested version 0.0.13, the issue remains.

```bash 
$ cargo webassembly run
   Compiling webassembly some-wasm package
thread 'main' panicked at 'could not copy: Os { code: 2, kind: NotFound, message: "No such file or directory" }', /home/gatewaynode/.cargo/registry/src/github.com-1ecc6299db9ec823/cargo-webassembly-0.0.13/src/main.rs:170:6
```

Might I suggest adding the `name` var in the `expect()` on line 170, as in `.expect(format!("could not copy: {:?}", name));` or switch to more comprehensive error handling on the file operations?

**Comment by richardanaya** [comment]
![image](https://user-images.githubusercontent.com/294042/100479483-3b711000-30a3-11eb-8584-e087e0ac3899.png)

hmm... here's my output for 0.0.14

maybe try 
```
cargo update
cargo install cargo-webassembly
```

wondering if this is OS specific hmm..

**Comment by gatewaynode** [comment]
I've been using the `cargo-update` crate to keep things up to date.

I think I figured it out, at least partly, compilation is failing silently during runtime so there is no artifact to copy.  When I manually copy the build options I get this error:
```
Thelio:~/code/some-wasm$ RUST_BACKTRACE=1 cargo build --target wasm32-unknown-unknown --release
   Compiling spin v0.5.2
   Compiling cty v0.2.1
   Compiling spin v0.7.0
   Compiling once_cell v1.5.2
   Compiling html_color v0.0.0
   Compiling memchr v2.3.4
   Compiling anystring v0.0.2
error[E0463]: can't find crate for `core`
  |
  = note: the `wasm32-unknown-unknown` target may not be installed

error: aborting due to previous error
```
I'm not sure yet why the wasm  toolchain didn't install, I've tried several time to add it with `rustup` and nothing changes.

**Comment by gatewaynode** [comment]
That was it, just needed the right target installed (might want to add this to the install or documentation).   I made 2 changes to get it to work, this first one is probably all that is necessary:
```
$ rustup target add wasm32-unknown-unknown
```
Also I added a `toolchain` file in the project root with one line `wasm32-unknown-unknown`.  This is apparently the new way to set per project target defaults.  The toolchain file is supposed to automatically make sure the target toolchain is installed before building, but this does not seem to be the case (https://github.com/rust-lang/rustup/issues/2575).  This new toolchain behavior is something to look out for in the future (https://blog.rust-lang.org/2020/11/27/Rustup-1.23.0.html).
